### PR TITLE
Add options to open links in default applications

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

This PR adds the ability to select default applications to open locations, tweets and generally links in. The currently supported applications are Google Maps for locations, Tweetbot and Twitterrific for tweets and Chrome and Firefox for other kinds of links ([#7159](https://wearezeta.atlassian.net/browse/ZIOS-7159)). 

The user can select these options under `Settings/Advanced`, but they will only be visible if the user has the option installed, there will be no option to select a default application for opening locations if the user only has Apple Maps for example. If the user has non of the mentioned apps the whole section will not be displayed.

If the opening fails (for example when the application that has been selected has ben removed from the device again) we fallback to open the link using Safari (or Apple Maps in case of locations).

### CODE REVIEW

File: Wire-iOS/Resources/Base.lproj/Localizable.strings [hunk 1]
```diff
@@ -485,6 +485,24 @@
...
 
+// Open in
+"self.settings.external_apps.header" = "External Apps";
+
+"self.settings.link_options.twitter.title" = "Tweets";
+"self.settings.link_options.maps.title" = "Maps";
+"self.settings.link_options.browser.title" = "Browser";
+
+"open_link.twitter.option.default" = "Safari / Twitter";
+"open_link.twitter.option.tweetbot" = "Tweetbot";
```
Comment (mikeger) [inline Wire-iOS/Resources/Base.lproj/Localizable.strings]: @raphaelrobert should this copy have (c) trademark attributes?

File: Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift [hunk 2]
```diff
@@ -220,6 +220,39 @@ class SettingsPropertyFactory {
             
             return SettingsBlockProperty(propertyName: propertyName, getAction: getAction, setAction: setAction)
 
+        case .tweetOpeningOption:
```
Comment (mikeger) [inline Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift]: I think you can simplify your code by putting the settings keys in `userDefaultsPropertiesToKeys`, then the thing maps it automagically. Less boilerplate!

Comment (daehn) [inline Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift]: Good point, will check it

File: Wire-iOS/Resources/Base.lproj/Localizable.strings (hunk 1 above)
Comment (mikeger) [inline Wire-iOS/Resources/Base.lproj/Localizable.strings]: @raphaelrobert

**Comment by daehn** [comment]
@mikeger @typfel updated.

**Pull request merged**